include/seqan3/alignment/matrix/alignment_trace_algorithms.hpp

```diff
@@ -119,8 +116,18 @@ alignment_trace(database_t && database,
     assert(row < matrix.rows());
     assert(col < matrix.cols());
 
-    std::deque<gapped_database_alphabet_t> gapped_database{};
-    std::deque<gapped_query_alphabet_t> gapped_query{};
+    auto begin_db = std::ranges::begin(database);
+    auto end_db = std::ranges::end(database);
+    auto begin_query = std::ranges::begin(query);
+    auto end_query = std::ranges::end(query);
+
+    alignment_t aligned_seq;
+    assign_unaligned(std::get<0>(aligned_seq),
+                     std::ranges::subrange<decltype(begin_db), decltype(end_db)>(begin_db, end_db));
```

Comment:
```suggestion
                     std::ranges::subrange<decltype(begin_db), decltype(end_db)>{begin_db, end_db});
```
also below

Reply:
```suggestion
                     view::all(database),
```

Also in the other places.

Reply:
There is a problem with view::all: There is no operator= for assigning gap_decorator<std::span> to gap_decorator<ranges::subrange>.

Is there an easy solution or should I stick to subrange?

Reply:
You need to change the gap decorator constructor to use our view::all and not std::view::all.